common/src/main/java/com/wynntils/core/notifications/MessageContainer.java

```diff
@@ -9,17 +9,29 @@
 
 public class MessageContainer {
     protected TextRenderTask message;
+    private TextRenderTask renderedMessage;
+    private int messageCount;
 
     public MessageContainer(String message) {
         this.message = new TextRenderTask(message, TextRenderSetting.DEFAULT);
     }
 
     public MessageContainer(TextRenderTask message) {
         this.message = message;
+        this.messageCount = 1;
+    }
+
+    public String getOriginalMessage() {
+        return message.getText();
     }
 
     public TextRenderTask getRenderTask() {
-        return message;
+        if (renderedMessage == null) return message;
```

Comment:
My suggestion is:
1) Rename it `currentMessage`, since it is not "rendered" yet

2) Set it directly in the constructor (to the original message) and you can just return it here, no need for a check.

Reply:
I sent you a ping in Discord on why the null-check was here and why it's not defined in the constructor because I had already tried that and was met with the following crash reason. https://hst.sh/ogusuragit Very possible that I was doing something incorrectly at the time of that comment, I believe that log is from the implementation at https://github.com/Wynntils/Artemis/pull/781/commits/ca26cf96a072d276347b35aae2d582ce006347ad.

Will rename it though.